airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionHandler.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.integrations.util
+
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.Command
+import io.airbyte.cdk.integrations.base.errors.messages.ErrorMessage
+import io.airbyte.commons.exceptions.ConfigErrorException
+import io.airbyte.commons.exceptions.ConnectionErrorException
+import io.airbyte.commons.exceptions.TransientErrorException
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.function.Consumer
+import kotlin.system.exitProcess
+import org.jetbrains.annotations.VisibleForTesting
+
+private val LOGGER = KotlinLogging.logger {}
+
+const val COMMON_EXCEPTION_MESSAGE_TEMPLATE: String =
+    "Could not connect with provided configuration. Error: %s"
+
+const val DATABASE_CONNECTION_ERROR: String =
+    "Encountered an error while connecting to the database error"
+
+enum class FailureType {
+    CONFIG,
+    TRANSIENT
+}
+
+data class ConnectorErrorProfile(
```

Comment:
Would it be possible to define a fluent/builder interface here?

Reply:
this is interesting, looking into this

Reply:
implemented the builder interface, thanks for this great suggestion! 

@postamar, could you also take a look? I think this aligns with your general comments above on having a syntax sugar. With the builder, it is much easier to understand the example in each connector's handler's error dictionary initialization.

Reply:
why do we need a builder when we can use named parameters?

Reply:
@stephane-airbyte I was not aware of this language feature. And yes, if the code is in Kotlin, we can directly use it. 

However, the code where we initialize the error dictionaries is Postgres and MySQL's exception handler class, currently written in Java, which doesn't support named parameters.

Reply:
@stephane-airbyte per our discussion, pushed a change which converted all java handler class and their test class into kotlin, and used named parameters to add error profiles.